codex/sales/states/proving.nim

```diff
@@ -1,29 +1,106 @@
+import std/options
 import pkg/chronicles
+import ../../clock
 import ../statemachine
 import ../salesagent
+import ../salescontext
 import ./errorhandling
-import ./filling
 import ./cancelled
 import ./failed
-import ./filled
+import ./errored
+import ./payout
 
 logScope:
     topics = "marketplace sales proving"
 
 type
+  SlotNotFilled* = object of Exception
```

Comment:
This should be at the very least `object of CatchableError`, but could also be `object of CodexError` (if importing from `../../errors`). `Exception` is used a base for both `CatchableError` and `Defect`, the latter of which is not catchable.

Reply:
Typical nim styling also suffixes with `Error`
```suggestion
  SlotNotFilledError* = object of CatchableError
```